### ISSUE TITLE
Display Attribute localization does not work on shared resource

### ISSUE THREAD

**Issue by akanmuratcimen** [aspnet/Mvc#5275]
In 1.1.0-alpha1-22125 localization Name property in DisplayAttribute works well when all models have their own resource files (assume that ~/Resources/Models/TestBindingModel.en.resx) but when I try to use single resource file by

```
.AddDataAnnotationsLocalization(options => {
    options.DataAnnotationLocalizerProvider = (type, factory) => 
        factory.Create(typeof(SharedResource));
})

```
localization does not work.

**Comment by danroth27** [comment]
@ryanbrandenburg

**Comment by ryanbrandenburg** [comment]
@akanmuratcimen I was able to reproduce this, I'm working on a fix now. Thanks!

**Comment by ryanbrandenburg** [comment]
Fixed by #5287. Thanks!
